sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,464 @@
+        payload = self.generate_data_from_schema(schema, seed)
+
+        key_record = {
+            "schema": schema,
+            "payload": payload
+        }
+        key_str = json.dumps(key_record)
+        return key_str
+
+    def stop(self) -> None:
+        self.termination_event.set()
+
+    def kill(self) -> None:
+        self.stop()
+
+    def verify_results(self) -> (list[dict | None], list[str | None]):
+        self.join(self.timeout)
+        errors = []
+        if self.undelivered_count > 0:
+            msg = f"{self.undelivered_count} records failed to be delivered."
+            TestFrameworkEvent(source=self.__class__.__name__, message=msg, severity=Severity.ERROR).publish()
+            errors.append(msg)
+        return [{'delivered_count': self.delivered_count}], errors
+
+
+class KafkaValidatorThread(KafkaThreadBase):
+    """
+    Thread to validate that records added to Kafka are correctly written to ScyllaDB.
+    """
+
+    def __init__(self, tester, params: SCTConfiguration, timeout: int, **kwargs):
+        """
+        Initializes the thread instance that validates ScyllaDB rows inserted by ScyllaDB Sink Connector.
+
+        :param tester: ClusterTester, the tester object instance.
+        :param params: SCTConfiguration, the test configuration parameters.
+        :param timeout: int, timeout for the thread execution in seconds.
+        :param kwargs: dict, additional keyword arguments for configuring the producer.
+
+        **Keyword Arguments:**
+        - record-format: str, format of records produced in Kafka.
+            Supported formats: 'json-with-schema', 'avro'.
+        - keyspace: str, name of the keyspace in ScyllaDB to validate records in.
+        - table: str, name of the table in ScyllaDB to validate records in.
+        - num-records: int, total number of records to validate in ScyllaDB.
+        - batch-size: int, number of records to validate at a time, to avoid reading the whole table at once.
+            Default is `500`.
+        - value-schema: str, JSON string representing the schema for the Kafka record value.
+            It is used to re-generate the value that was written to Kafka (this way do not need to re-read
+            expected value from Kafka for validation).
+
+                **Usage Example in Test Configuration:**
+        ```
+        stress_read_cmd: >
+            python_thread -thread=KafkaValidatorThread
+            -keyspace=keyspace1 -table=table1 -record-format=json-with-schema -num-records=1000 -batch-size=500
+            -value-schema='{"type":"struct","fields":[{"type":"string","optional":false,"field":"name"},{"type":"string","optional":true,"field":"department"}]}'
+        ```
+        """
+        super().__init__(tester, params, timeout, **kwargs)
+        self.mismatches = []
+        self.batch_size = int(kwargs.get('batch-size', 500))
+        self.keyspace = kwargs.get('keyspace', None)
+        self.table = kwargs.get('table', None)
```

Comment:
after stripping off kafka arguments it's good to verify all the arguments if are matching required ones. This way we would prevent silently passing if someone makes typo in them. E.g. if someone passes `-num-rcords` then we would use default 1000 instead of selected one and no one would know.